File: php/v3.0/examples/Advertisers/CreateAdvertiserLandingPage.php
<?php
/*
 * Copyright 2017 Google Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

// Require the base class.
require_once dirname(__DIR__) . '/BaseExample.php';

/**
 * This example creates an advertiser landing page.
 */
class CreateAdvertiserLandingPage extends BaseExample {
  /**
   * (non-PHPdoc)
   * @see BaseExample::getInputParameters()
   * @return array
   */
  protected function getInputParameters() {
    return [['name' => 'user_profile_id',
             'display' => 'User Profile ID',
             'required' => true],
            ['name' => 'advertiser_id',
             'display' => 'Advertiser ID',
             'required' => true],
            ['name' => 'landing_page_name',
             'display' => 'Landing Page Name',
             'required' => true],
            ['name' => 'landing_page_url',
             'display' => 'Landing Page URL',
             'required' => true]];
  }

  /**
   * (non-PHPdoc)
   * @see BaseExample::run()
   */
  public function run() {
    $values = $this->formValues;

    printf(
        '<h2>Creating advertiser landing page for advertiser %d</h2>',
        $values['advertiser_id']
    );

    $landingPage = new Google_Service_Dfareporting_LandingPage();
    $landingPage->setAdvertiserId($values['advertiser_id']);
    $landingPage->setArchived(false);
    $landingPage->setName($values['landing_page_name']);
    $landingPage->setUrl($values['landing_page_url']);

    $result = $this->service->advertiserLandingPages->insert(
        $values['user_profile_id'], $landingPage);

    $this->printResultsTable('Advertiser landing page created.', [$result]);
  }

  /**
   * (non-PHPdoc)
   * @see BaseExample::getName()
   * @return string
   */
  public function getName() {
    return 'Create Advertiser Landing Page';
  }

  /**
   * (non-PHPdoc)
   * @see BaseExample::getResultsTableHeaders()
   * @return array
   */
  public function getResultsTableHeaders() {
    return ['id' => 'Advertiser Landing Page ID',
            'name' => 'Advertiser Landing Page Name',
            'url' => 'Advertiser Landing Page URL'];
  }
}

File: java/v3.3/src/main/java/com/google/api/services/samples/dfareporting/advertisers/CreateAdvertiserLandingPage.java
// Copyright 2017 Google Inc. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package com.google.api.services.samples.dfareporting.advertisers;

import com.google.api.services.dfareporting.Dfareporting;
import com.google.api.services.dfareporting.model.LandingPage;
import com.google.api.services.samples.dfareporting.DfaReportingFactory;

/**
 * This example creates an advertiser landing page.
 */
public class CreateAdvertiserLandingPage {
  private static final String USER_PROFILE_ID = "INSERT_USER_PROFILE_ID_HERE";

  private static final String ADVERTISER_ID = "INSERT_ADVERTISER_ID_HERE";
  private static final String LANDING_PAGE_NAME = "INSERT_LANDING_PAGE_NAME_HERE";
  private static final String LANDING_PAGE_URL = "INSERT_LANDING_PAGE_URL_HERE";

  public static void runExample(Dfareporting reporting, long profileId, long advertiserId,
      String landingPageName, String landingPageUrl) throws Exception {
    // Create the landing page structure.
    LandingPage landingPage = new LandingPage();
    landingPage.setAdvertiserId(advertiserId);
    landingPage.setArchived(false);
    landingPage.setName(landingPageName);
    landingPage.setUrl(landingPageUrl);

    // Create the landing page
    LandingPage result =
        reporting.advertiserLandingPages().insert(profileId, landingPage).execute();

    // Display the landing page ID.
    System.out.printf("Advertiser landing page with ID %d and name \"%s\" was created.%n",
        result.getId(), result.getName());
  }

  public static void main(String[] args) throws Exception {
    Dfareporting reporting = DfaReportingFactory.getInstance();

    long profileId = Long.parseLong(USER_PROFILE_ID);
    long advertiserId = Long.parseLong(ADVERTISER_ID);

    runExample(reporting, profileId, advertiserId, LANDING_PAGE_NAME, LANDING_PAGE_URL);
  }
}
